app/javascript/packs/application.js

```diff
@@ -11,3 +11,29 @@ import "channels"
 Rails.start()
 Turbolinks.start()
 ActiveStorage.start()
+
+window.addEventListener("load", () => {
+  const newConfigButton = document.getElementById("show_new_config_form");
+  const newSourceButton = document.getElementById("show_new_source_form");
+  const sourceForm = document.getElementById("kernel_source_form");
+  const configForm = document.getElementById("kernel_config_form");
+
+  sourceForm.style.display = "none";
+  configForm.style.display = "none";
+
+  newConfigButton.addEventListener("click", (e) => {
+    if( configForm.style.display == "none"){
+      configForm.style.display = "block";
+    } else {
+      configForm.style.display = "none";
+    }
+  });
+
+  newSourceButton.addEventListener("click", (e) => {
+    if( sourceForm.style.display == "none"){
+      sourceForm.style.display = "block";
+    } else {
+      sourceForm.style.display = "none";
+    }
```

Comment:
This is a lotta duplication.  How about we add a function `toggleFormVisibility(form, target)` that we can call, passing the two elements?  That way if we change the logic at all, it's consolidated into one place.

Reply:
Fixed in [760c5c1](https://github.com/Brian-Demon/kernel-builder/pull/11/commits/760c5c14b4fcb504704539728bb0f797cfb47e94)